src/en/kmanga/src/eu/kanade/tachiyomi/extension/en/kmanga/KManga.kt

```diff
@@ -0,0 +1,369 @@
+package eu.kanade.tachiyomi.extension.en.kmanga
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.POST
+import eu.kanade.tachiyomi.network.asObservable
+import eu.kanade.tachiyomi.network.interceptor.rateLimitHost
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonArray
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.jsonArray
+import kotlinx.serialization.json.jsonObject
+import kotlinx.serialization.json.jsonPrimitive
+import okhttp3.FormBody
+import okhttp3.Headers
+import okhttp3.HttpUrl
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import okio.ByteString.Companion.encodeUtf8
+import org.jsoup.nodes.Element
+import rx.Observable
+import uy.kohesive.injekt.injectLazy
+import java.net.URLDecoder
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class KManga : HttpSource() {
+
+    override val name = "K Manga"
+    override val baseUrl = "https://kmanga.kodansha.com"
+    override val lang = "en"
+    override val supportsLatest = false
+
+    private val apiUrl = "https://api.kmanga.kodansha.com"
+    private val dateFormat = SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.US)
+    private val json: Json by injectLazy()
+
+    override fun headersBuilder(): Headers.Builder = Headers.Builder()
```

Comment:
```suggestion
    override fun headersBuilder(): Headers.Builder = super.headersBuilder()
```
to correctly set the useragent